gr-blocks/lib/correctiq_auto_impl.cc

```diff
@@ -0,0 +1,123 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "correctiq_auto_impl.h"
+#include <gnuradio/io_signature.h>
+
+namespace gr {
+namespace blocks {
+
+correctiq_auto::sptr
+correctiq_auto::make(float samp_rate, float freq, float gain, float syncWindow)
+{
+    return gnuradio::get_initial_sptr(
+        new correctiq_auto_impl(samp_rate, freq, gain, syncWindow));
+}
+
+/*
+ * The private constructor
+ */
+correctiq_auto_impl::correctiq_auto_impl(float samp_rate,
+                                         float freq,
+                                         float gain,
+                                         float syncWindow)
+    : gr::sync_block("correctiq_auto",
+                     gr::io_signature::make(1, 1, sizeof(gr_complex)),
+                     gr::io_signature::make(1, 1, sizeof(gr_complex)))
+{
+    avg_real = 0.0;
+    avg_img = 0.0;
+
+    d_samp_rate = samp_rate;
+    d_freq = freq;
+    d_gain = gain;
+    d_syncWindow = syncWindow;
+
+    synchronized = false;
+
+    std::cout << "CorrectIQ Auto Synchronizing..." << std::endl;
```

Comment:
ah! Could you use replace that with 

```
GR_LOG_INFO(d_logger,  "Auto Synchronizing..."); 
```

That way, you get configurable logging. 

(the logger should inherently also print the block's name)